From 4d9c0c04508678592789cbb29b2a80c99caa2d47 Mon Sep 17 00:00:00 2001
From: Wei Yongjun <yongjun_wei@trendmicro.com.cn>
Date: Fri, 22 Mar 2013 09:11:00 -0600
Subject: [PATCH 324/858] cciss: fix invalid use of sizeof in
 cciss_find_cfgtables()

sizeof() when applied to a pointer typed expression gives the
size of the pointer, not that of the pointed data.

Signed-off-by: Wei Yongjun <yongjun_wei@trendmicro.com.cn>
Acked-by: scameron@beardog.cce.hp.com
Signed-off-by: Jens Axboe <axboe@kernel.dk>
(cherry picked from commit d2b805d89510737ea80c1469f854a16480d19778)

Signed-off-by: Clark Williams <williams@redhat.com>
Signed-off-by: John Kacur <jkacur@redhat.com>
---
 drivers/block/cciss.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/block/cciss.c b/drivers/block/cciss.c
index ade58bc..1c1b8e5 100644
--- a/drivers/block/cciss.c
+++ b/drivers/block/cciss.c
@@ -4206,7 +4206,7 @@ static int cciss_find_cfgtables(ctlr_info_t *h)
  if (rc)
   return rc;
  h->cfgtable = remap_pci_mem(pci_resource_start(h->pdev,
-  cfg_base_addr_index) + cfg_offset, sizeof(h->cfgtable));
+  cfg_base_addr_index) + cfg_offset, sizeof(*h->cfgtable));
  if (!h->cfgtable)
   return -ENOMEM;
  rc = write_driver_ver_to_cfgtable(h->cfgtable);
-- 
1.8.3.1